app/sdil/controllers/variation/CopackForOthersController.scala

```diff
@@ -16,32 +16,33 @@
 
 package sdil.controllers.variation
 
-import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.i18n.MessagesApi
 import play.api.mvc.{Action, AnyContent}
 import sdil.actions.VariationAction
 import sdil.config.AppConfig
 import sdil.controllers.RadioFormController
 import uk.gov.hmrc.http.cache.client.SessionCache
-import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import views.html.softdrinksindustrylevy.register.radio_button
 
-import scala.concurrent.Future
-
 class CopackForOthersController(val messagesApi: MessagesApi,
-                                cache: SessionCache,
+                                val cache: SessionCache,
                                 variationAction: VariationAction)
                                (implicit config: AppConfig)
-  extends FrontendController with I18nSupport {
+  extends Journey {
 
-  def show: Action[AnyContent] = variationAction { implicit request =>
+  def show: Action[AnyContent] = variationAction.async { implicit request =>
     val filledForm = RadioFormController.form.fill(request.data.copackForOthers)
-
-    Ok(radio_button(filledForm, "packageCopack", backLink, submitAction))
+    backLink(routes.CopackForOthersController.show()) map { link =>
+      Ok(radio_button(filledForm, "packageCopack", link, submitAction))
+    }
   }
 
   def submit: Action[AnyContent] = variationAction.async { implicit request =>
     RadioFormController.form.bindFromRequest().fold(
-      errors => Future.successful(BadRequest(radio_button(errors, "packageCopack", backLink, submitAction))),
+      errors =>
+        backLink(routes.CopackForOthersController.show()) map { link =>
```

Comment:
is it necessary to save the back link on every form POST?

it looks to me like this will result in a lot of unnecessary calls to keystore